nntrainer/graph/network_graph.cpp

```diff
@@ -313,6 +313,12 @@ int NetworkGraph::checkInitializedGraph() {
     if (lnode->getTrainable() ||
         must_support_backwarding.find(lnode->getName()) !=
           must_support_backwarding.end()) {
+      lnode->needsCalcGradient(true);
+#ifdef ENABLE_TEST
+      if (lnode->supportBackwarding() && !optimize_memory) {
+        lnode->needsCalcDerivative(true);
```

Comment:
This function name suggest to check and not to touch the node property. If it is now, how about split this to different function or change the function name?

Reply:
There are two needsCalcDerivative(). One is a getter, one is a setter. Do you want different names for the two?

Reply:
no it's about `checkInitializedGraph()` seems not clear to me.

it sounds like a sanity check but actually writing something

Reply:
I understand, I will split it into two functions.

Reply:
renamed to `markNodesForBackwarding`.